bug(jxa): listFolders with parentId returns empty — nested folder filter broken

## Bug

`listFolders` with a `parentId` filter returns an empty array `[]` even when sub-folders exist under that parent.

## Repro

Integration test: `listFolders filters by parentId (nested folders)` fails:
```
AssertionError: expected [] to deeply equal ['evRUfl7Y8RA']
```

## Root cause

When creating a sub-folder via `folder_create.js` with `parentId`, the folder is created inside the parent. But `folder_list.js` may be filtering by `folder.parentId === args.parentId` incorrectly (e.g., using `flattenedFolders()` but not correctly reading the `parentId` from the JXA object), or the `buildFolder` function doesn't correctly extract `parentId` for sub-folders created via the push pattern.

## Acceptance Criteria

- [ ] `folder_create.js` with `parentId` creates the sub-folder correctly (verify via `getFolder`)
- [ ] `folder_list.js` with `parentId` returns sub-folders of that parent
- [ ] Integration test `listFolders filters by parentId (nested folders)` passes

--- a/src/scripts/jxa/folder_list.js
+++ b/src/scripts/jxa/folder_list.js
@@ -14,15 +14,29 @@ function run(argv) {
   const ofApp = Application("OmniFocus");
   ofApp.includeStandardAdditions = false;
 
-  function buildFolder(folder) {
-    let parentId = null;
+  // OmniFocus 4.8.8: folder.parent() throws "Can't convert types." for
+  // sub-folders. Build a reverse map (childId → parentId) by iterating each
+  // folder's .folders() children instead — that API works correctly.
+  const allFolders = ofApp.defaultDocument.flattenedFolders();
+  const parentMap = {};
+  for (let i = 0; i < allFolders.length; i++) {
     try {
-      const p = folder.parent();
-      if (p && p.class() !== "document") parentId = p.id();
+      const subs = allFolders[i].folders();
+      const pid = allFolders[i].id();
+      for (let j = 0; j < subs.length; j++) {
+        try {
+          parentMap[subs[j].id()] = pid;
+        } catch (_e) {}
+      }
     } catch (_e) {}
+  }
+
+  function buildFolder(folder) {
+    const id = folder.id();
+    const parentId = parentMap[id] !== undefined ? parentMap[id] : null;
 
     return {
-      id: folder.id(),
+      id: id,
       name: folder.name(),
       parentId: parentId,
       projectCount: folder.projects ? folder.projects().length : 0,
@@ -36,9 +50,7 @@ function run(argv) {
     };
   }
 
-  const allFolders = ofApp.defaultDocument.flattenedFolders();
   const result = [];
-
   for (let i = 0; i < allFolders.length; i++) {
     const built = buildFolder(allFolders[i]);
     if (args.parentId !== undefined && built.parentId !== args.parentId) continue;

